### PR TITLE
SNOW-944715 Validate account domain

### PR DESCRIPTION
**Pull request by sfc-gh-knozderko**
### Description
Validate account domain to prevent SSRF.

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data/Core/Session/SFSessionProperty.cs [hunk 1]
```diff
@@ -101,17 +102,19 @@ class SFSessionPropertyAttr : Attribute
...
+        private static List<SFSessionProperty> secretProps =
             new List<SFSessionProperty>{
                 SFSessionProperty.PASSWORD,
                 SFSessionProperty.PRIVATE_KEY,
                 SFSessionProperty.TOKEN,
                 SFSessionProperty.PRIVATE_KEY_PWD,
                 SFSessionProperty.PROXYPASSWORD,
             };
+        
+        private const string AccountRegexString = "^\\w[\\w.-]+\\w$";
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data/Core/Session/SFSessionProperty.cs:117]: what about underscore ? https://docs.snowflake.com/en/sql-reference/sql/create-account#required-parameters

Comment (sfc-gh-knozderko) [reply Snowflake.Data/Core/Session/SFSessionProperty.cs:117]: underscore is already covered by \w pattern

**Review by sfc-gh-dstempniak — APPROVED** [review]

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Review by sfc-gh-mhofman — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs [hunk 2]
```diff
@@ -29,7 +29,10 @@ public void TestThatPropertiesAreParsed(TestCase testcase)
         [Test]
         [TestCase("ACCOUNT=testaccount;USER=testuser;PASSWORD=testpassword;FILE_TRANSFER_MEMORY_THRESHOLD=0;", "Error: Invalid parameter value 0 for FILE_TRANSFER_MEMORY_THRESHOLD")]
         [TestCase("ACCOUNT=testaccount;USER=testuser;PASSWORD=testpassword;FILE_TRANSFER_MEMORY_THRESHOLD=xyz;", "Error: Invalid parameter value xyz for FILE_TRANSFER_MEMORY_THRESHOLD")]
-        public void TestThatItFailsForWrongFileTransferMaxBytesInMemoryParameter(string connectionString, string expectedErrorMessagePart)
+        [TestCase("ACCOUNT=testaccount?;USER=testuser;PASSWORD=testpassword", "Error: Invalid parameter value testaccount? for ACCOUNT")]
+        [TestCase("ACCOUNT=complicated.long.testaccount?;USER=testuser;PASSWORD=testpassword", "Error: Invalid parameter value complicated.long.testaccount? for ACCOUNT")]
+        [TestCase("ACCOUNT=?testaccount;USER=testuser;PASSWORD=testpassword", "Error: Invalid parameter value ?testaccount for ACCOUNT")]
```
[the remark this answers is not in the document]
Comment (sfc-gh-knozderko) [reply Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs:34]: done

File: Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs [hunk 3]
```diff
@@ -40,6 +43,20 @@ public void TestThatItFailsForWrongFileTransferMaxBytesInMemoryParameter(string
             Assert.AreEqual(SFError.INVALID_CONNECTION_PARAMETER_VALUE.GetAttribute<SFErrorAttr>().errorCode, exception.ErrorCode);
             Assert.IsTrue(exception.Message.Contains(expectedErrorMessagePart));
         }
+
+        [Test]
+        [TestCase("ACCOUNT=;USER=testuser;PASSWORD=testpassword")]
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs:48]: That solves a thing pointed by  SNYK , thx

Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs:48]: Thx, that solves a problem from SNYK

**Review by sfc-gh-mhofman — APPROVED** [review]

**Pull request merged**